### ISSUE TITLE
`make test` fails if a file named `x` is in the git repo

### ISSUE THREAD

**Issue by krader1961** [fish-shell/fish-shell#3852]
If a file whose name is a single character exists in the *tests* directory of the fish-shell repo then `make test` fails the *fish_tests* `illegal command exit code` test. This issue is to investigate whether we can make the tests more hermetic and less prone to being broken by random files in a local repo.

I noticed this when I created a file named `x` in the top-level directory of my git repo while exploring the problem someone reported that `echo -n x` didn't work as expected. The file content was the result of the `script x` command. Note that the fact I created a file named `x` is merely coincidence. If I rename the file to `y` or `a` I get the same unexpected test failure.

I then did `make test` and was surprised that the test failed in this fashion:

```
Testing illegal command exit code
/Users/krader/projects/3rd-party/fish-shell
Error: command '?': expected exit code 123 , got 127
```

Note that an empty file doesn't cause a problem. It has to contain some text that isn't valid fish script. For example, `Script started on Tue Feb 14 09:34:38 2017`.

**Comment by krader1961** [comment]
Bisected to commit bb686a2236 by @ridiculousfish so I'll trade you this for the PROMPT_SP bug I introduced :smile:

Also, the single character file name needs to be in the repo root, not the *tests* directory.

**Comment by ridiculousfish** [comment]
Nice find. This could have caused a lot of headaches later.

**Comment by krader1961** [comment]
I have started working on fixing this issue. The fix should have been straightforward. But the low level tests implemented by *fish_tests.cpp* have a number of surprising failure modes if the completion target is not an absolute path. I have fixed all but one of those test failures. It is not clear if the remaining failure represents a bug in the *fish_tests.cpp* program or a bug in the code being tested.
